### PR TITLE
Add optional `writeToStreamMulti` function to the World interface

### PR DESCRIPTION
**Pull request by TooTallNate**
Added an optional `writeToStreamMulti` function to the World interface to optimize batch writing of multiple chunks to a stream.

This is an alternative solution to the issue reported in #764.

### What changed?

- Added an optional `writeToStreamMulti` method to the `Streamer` interface in the World API
- Implemented buffering in `WorkflowServerWritableStream` to batch chunks before flushing
- Added implementations of `writeToStreamMulti` in all world providers:
  - `world-local`: Writes multiple chunks in parallel while preserving order
  - `world-postgres`: Performs a batch insert for all chunks
  - `world-vercel`: Encodes multiple chunks into a length-prefixed binary format
- Added a constant `STREAM_FLUSH_INTERVAL_MS` (10ms) to control buffer flush timing

### Why make this change?

This optimization reduces network overhead when writing many small chunks to a stream by batching them together. This is particularly beneficial for:

1. The Vercel world implementation where each write previously required a separate HTTP request
2. The Postgres world implementation where batch inserts are more efficient
3. Any scenario with high-frequency, small writes to streams

The implementation gracefully falls back to sequential writes for world implementations that don't support the new method.

### CODE REVIEW

**Comment by TooTallNate** [comment]
* **#867** <a href="https://app.graphite.com/github/pr/vercel/workflow/867?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/vercel/workflow/867?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/world-vercel/src/utils.ts [hunk 1]
```diff
@@ -14,7 +14,8 @@ import { version } from './version.js';
  *
  * Example: 'https://workflow-server-git-branch-name.vercel.sh'
  */
-const WORKFLOW_SERVER_URL_OVERRIDE = '';
+const WORKFLOW_SERVER_URL_OVERRIDE =
+  'https://workflow-server-git-01-27-implementmultiwritemodeforstre-1e1eee.vercel.sh';
```
Comment (pranaygp) [inline packages/world-vercel/src/utils.ts]: ⚠️ **Must fix**: This hardcoded server URL override needs to be reverted to an empty string before merging. This is likely causing the "No Test Overrides" CI failure.

File: packages/core/src/serialization.ts [hunk 2]
```diff
@@ -173,12 +179,69 @@ export class WorkflowServerWritableStream extends WritableStream<Uint8Array> {
...
+          await flushPromise;
+          flushPromise = null;
+        }
+
+        buffer.push(chunk);
+        scheduleFlush();
       },
       async close() {
+        // Wait for any in-progress flush to complete
+        if (flushPromise) {
```
Comment (pranaygp) [inline packages/core/src/serialization.ts:355]: ⚠️ **Should fix**: The WritableStream doesn't implement an `abort()` handler. If the stream is aborted mid-write, the `flushTimer` won't be cleaned up (timer leak) and buffered chunks could be lost silently.

Consider adding:
```typescript
super({
  async write(chunk) { ... },
  async close() { ... },
  async abort() {
    if (flushTimer) {
      clearTimeout(flushTimer);
      flushTimer = null;
    }
    buffer = [];
  }
});
```

File: packages/core/src/serialization.ts [hunk 3]
```diff
@@ -173,12 +179,69 @@ export class WorkflowServerWritableStream extends WritableStream<Uint8Array> {
...
+
+    const scheduleFlush = (): void => {
+      if (flushTimer) return; // Already scheduled
+
+      flushTimer = setTimeout(() => {
+        flushTimer = null;
+        flushPromise = flush();
+      }, STREAM_FLUSH_INTERVAL_MS);
+    };
+
```
Comment (pranaygp) [inline packages/core/src/serialization.ts:341]: Nice to have: The sequence here sets `flushPromise = null` before the await fully settles. This could theoretically allow a new flush to start while the previous one's side effects are still settling. Might be fine in practice, but worth double-checking.

File: packages/world-postgres/src/streamer.ts [hunk 4]
```diff
@@ -114,6 +118,44 @@ export function createStreamer(postgres: Sql, drizzle: Drizzle): Streamer {
...
+          chunkId: chunkIds[i],
+          streamId: name,
+          runId,
+          chunkData: toBuffer(chunk),
+          eof: false,
+        }))
+      );
+
+      // Notify for each chunk (could be batched in future if needed)
+      for (const chunkId of chunkIds) {
```
Comment (pranaygp) [inline packages/world-postgres/src/streamer.ts:147]: Nice to have: As noted, this still generates N notifications for N chunks. The batch insert is the bigger win, but if notifications become a bottleneck in the future, consider batching these as well (e.g., single notification with array of chunk IDs).

File: packages/core/src/serialization.ts [hunk 5]
```diff
@@ -158,6 +158,12 @@ export class WorkflowServerReadableStream extends ReadableStream<Uint8Array> {
   }
 }
 
+/**
+ * Default flush interval in milliseconds for buffered stream writes.
+ * Chunks are accumulated and flushed together to reduce network overhead.
+ */
```
Comment (pranaygp) [inline packages/core/src/serialization.ts:278]: ⚠️ **Should add**: This new buffering logic adds significant complexity (flush timing, buffer state, promise handling) but lacks unit tests. Consider adding tests for:
- Buffer flush timing behavior
- Handling of concurrent writes
- Edge cases with rapid writes
- Behavior when `writeToStreamMulti` is/isn't available

File: packages/world-vercel/src/streamer.ts [hunk 6]
```diff
@@ -14,6 +14,41 @@ function getStreamUrl(
   return new URL(`${httpConfig.baseUrl}/v2/stream/${encodeURIComponent(name)}`);
 }
 
+/**
```
Comment (pranaygp) [inline packages/world-vercel/src/streamer.ts:17]: Nice to have: Consider adding unit tests for `encodeMultiChunks` - the length-prefixed binary format is non-trivial and would benefit from tests verifying correct encoding, especially for edge cases like empty chunks or very large chunks.
